HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1636,41 +1561,105 @@ def self.apply_infiltration_and_ventilation_fans(model, weather, vent_mech, vent
+    infil_program.addLine('Set Qexhaust = Qrange+Qbath')
+    infil_program.addLine('Set Qsupply = 0')
+    if (not vent_mech.nil?)
+      if [HPXML::MechVentTypeExhaust].include? vent_mech.fan_type
+        infil_program.addLine('Set Qexhaust = Qexhaust+QWHV')
+      elsif [HPXML::MechVentTypeSupply, HPXML::MechVentTypeCFIS].include? vent_mech.fan_type
+        infil_program.addLine('Set Qsupply = Qsupply+QWHV')
+      elsif [HPXML::MechVentTypeBalanced, HPXML::MechVentTypeERV, HPXML::MechVentTypeHRV].include? vent_mech.fan_type
+        infil_program.addLine('Set Qexhaust = Qexhaust+QWHV')
+        infil_program.addLine('Set Qsupply = Qsupply+QWHV')
+      end
     end
 
-    infil_program.addLine('Set Q_tot_flow = (((Qu^2)+(Qn^2))^0.5)')
-    infil_program.addLine('Set Q_tot_flow = (@Max Q_tot_flow 0)')
+    # Calculate adjusted infiltration based on mechanical ventilation system
+    infil_program.addLine('Set Qfan = (@Max Qexhaust Qsupply)')
+    if Constants.ERIVersions.index(@eri_version) >= Constants.ERIVersions.index('2019')
+      # Follow ASHRAE 62.2-2016, Normative Appendix C equations for time-varying total airflow
+      infil_program.addLine('If Qfan > 0')
+      # Balanced system if the total supply airflow and total exhaust airflow are within 10% of their average.
+      infil_program.addLine('  Set Qavg = ((Qexhaust + Qsupply) / 2.0)')
+      infil_program.addLine('  If ((@Abs (Qexhaust - Qavg)) / Qavg) <= 0.1') # Only need to check Qexhaust, Qsupply will give same result
+      infil_program.addLine('    Set phi = 1')
+      infil_program.addLine('  Else')
+      infil_program.addLine('    Set phi = (Qinf / (Qinf + Qfan))')
+      infil_program.addLine('  EndIf')
+      infil_program.addLine('  Set Qinf_adj = phi * Qinf')
+      infil_program.addLine('Else')
+      infil_program.addLine('  Set Qinf_adj = Qinf')
+      infil_program.addLine('EndIf')
+    else
+      infil_program.addLine('Set Qu = (@Abs (Qexhaust - Qsupply))') # Unbalanced flow
+      infil_program.addLine('Set Qb = Qfan - Qu') # Balanced flow
+      infil_program.addLine('Set Qtot = (((Qu^2) + (Qinf^2)) ^ 0.5) + Qb')
+      infil_program.addLine('Set Qinf_adj = Qtot - Qu - Qb')
+    end
 
     # Assign total airflow to different component loads
-    infil_program.addLine("Set #{imbal_mechvent_flow_actuator.name} = (@Abs ((Qout - QductsOut) - (Qin - QductsIn)))")
-    infil_program.addLine("Set #{imbal_ducts_flow_actuator.name} = (@Abs (QductsOut - QductsIn))")
-
-    # Assign the remainder to infiltration:
-    infil_program.addLine("Set #{infil_flow_actuator.name} = Q_tot_flow - #{imbal_mechvent_flow_actuator.name} - #{imbal_ducts_flow_actuator.name}")
-    infil_program.addLine("If #{infil_flow_actuator.name} < 0")
-    infil_program.addLine("  Set #{infil_flow_actuator.name} = 0")
-    infil_program.addLine('EndIf')
+    if (not vent_mech.nil?) && [HPXML::MechVentTypeERV, HPXML::MechVentTypeHRV].include?(vent_mech.fan_type)
```

Comment:
It is, because I ended up moving Balanced mech vent systems (or the balanced portion of, e.g., a supply system w/ exhaust bath/kitchen fans) out of the ERV/HRV EMS load program for simplicity. I did a quick test and found that change alone has a tiny impact on simulation results. 

Do you recall why we originally decided to include balanced systems in the EMS load program?

Reply:
The balanced system was not included when we first implemented ERV/HRV into EMS program, but nor was the balanced airflow added into infiltration as we did now(just unbalanced airflow added). Tracking back to where it first came into the place, it's added in component_loads branch (probably as a handy bugfix?).